enhancements/network/communication-flows-matrix-ingress.md

```diff
@@ -0,0 +1,250 @@
+---
+title: communication-flows-matrix-ingress
+authors:
+  - "@sabinaaledort"
+  - "@yuvalk"
+  - "@liornoy"
+reviewers:
+  - "@cybertron"
+  - "@trozet"
+  - "@wking"
+  - "@danwinship"
+  - "@msherif1234"
+approvers:
+  - "@cybertron"
+  - "@trozet"
+  - "@wking"
+  - "@danwinship"
+  - "@msherif1234"
+api-approvers:
+  - None
+creation-date: 2023-03-07
+last-updated: 2023-03-07
+tracking-link:
+  - https://issues.redhat.com/browse/TELCOSTRAT-77
+---
+
+# Communication ingress flows matrix of OpenShift and Operators
+
+## Summary
+
+This enhancement allows to automatically generate an accurate and up-to-date 
+communication flows matrix that can be delivered to customers as part of 
+product documentation for all ingress flows of OpenShift (multi-node and
+single-node deployments) and Operators.
+
+## Motivation
+
+Security-conscious customers need OpenShift flows matrix for regulatory reasons 
+and/or to implement firewall rules to restrict traffic to the minimum set of
+required flows only, on-node firewall or external.
+
+### User Stories
+
+- As an OpenShift cluster administrator, I want documentation on the expected 
+  flows of traffic incoming to every OpenShift installation so I can set up 
+  firewall rules such as nftables, NGFW, etc. to restrict traffic to the 
+  minimum required set of flows only.
+
+- As an OpenShift cluster administrator, I want a "one-click" feature to 
+  generate an accurate, relevant, matrix for the running OpenShift cluster.
+
+- As a testing engineer, I want a machine-readable ingress traffic matrix 
+  so that I can analyze and compare with tools such as nmap.
+
+### Goals
+
+- Provide a mechanism to automatically generate an accurate and up-to-date 
+  OpenShift communication ingress flows matrix.
+
+- Keep the communication matrix documented in OpenShift release documents
+  updated and validate it does not change.
+
+### Non-Goals
+
+- Egress traffic.
+
+## Proposal
+
+We introduce a new module to generate an accurate and up-to-date OpenShift 
+communication ingress flows matrix.
+
+- A communication matrix describing the expected flows of incoming traffic will 
+  be included in every OpenShift release documentation.
+
+- A new `oc` command will be added to generate a current snapshot of known 
+  listening ports in a running cluster, `oc adm communication-matrix generate`.
+
+- A new option will be added to OpenShift web console to generate an up-to-date
+  communication matrix.
```

Comment:
How will that be different from what the network observability features provide?

cc @stleerh

Reply:
we are targeting a declarative approach whereas observability matrix is looking into what is actually running.
IMHO - what is not declared is "wrong" and should be blocked.
observability feature might be used to detect those and correct them but still need also a tool to extract the declared matrix